frontend/src/pages/Home.jsx

```diff
@@ -1,60 +1,63 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faGithub } from "@fortawesome/free-brands-svg-icons";
-import { useEffect } from "react";
-import { useNavigate } from "react-router-dom";
+import { faGithub } from '@fortawesome/free-brands-svg-icons';
+import { useEffect, useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import '../styles/home.css';
+import { UserContext } from '../App';
 
-const AUTH_ENDPOINT = "https://github.com/login/oauth/authorize"
-const data = {
-  "client_id": process.env.REACT_APP_CLIENT_ID,
-  "redirect_uri": "http://localhost:3000/",
-  "scope": ["repo"]
-}
-
-export default function Home() {
+export default function Home () {
+  const { user, setUser } = useContext(UserContext);
   const navigate = useNavigate();
   useEffect(() => {
-    let token = localStorage.getItem("token");
-    if (token) {
-      navigate("/search");
-      return
-    }
-    let qs = window.location.search;
-    let params = new URLSearchParams(qs);
-    let code = params.get("code");
-    if (code && !token){
-      console.log("calling backend!!");
-      fetch("http://localhost:3000/getToken?"+params.toString()).then((response) =>{
-        let data = response.json()
-        console.log(data);
-        return data;
+    if (user === null) {
+      fetch('http://localhost:3000/auth', {
+        credentials: 'include'
       }).then((res) => {
-        console.log('this is res '+res);
-        if (res.access_token){
-          localStorage.setItem("token", res.access_token);
+        if (res.status === 200) {
+          setUser({});
+          navigate('/search');
         }
+      });
+    }
+
+    const qs = window.location.search;
+    const params = new URLSearchParams(qs);
+    const code = params.get('code');
+    console.log('got code', code);
+    if (code) {
+      fetch('http://localhost:3000/token?' + params.toString(),
+        {
+          credentials: 'include'
+        }).then(() => {
+        setUser({});
+        navigate('/search');
       }).catch((e) => {
         console.log(e);
-      })
+      });
     }
-  }, [])
-  const searchParams = new URLSearchParams(data);
+  }, []);
+
+  function Login () {
+    fetch('http://localhost:3000/login').then((res) => {
+      if (res.status === 200) {
+        return res.json();
+      }
+    }).then(data => {
+      window.location.assign(data.url);
```

Comment:
looks like you're using react-router-dom. It should have a `useRouter` hook that you can use to redirect the user.